### PR TITLE
Profile from template

### LINKED ISSUE #162: Create ServerProfile from Template
### Scenario/Intent
I'd like to be able to define a `oneview_server_profile` resource that uses a ServerProfileTemplate as a base config (but allows me to override some of those settings with properties I set in addition). This would probably be done by adding a new `template` or `server_profile_template` property.

### Environment Details
 - **oneview Cookbook Version:** 1.1.0
 - **OneView Version:** 2.0 and 3.0

### PR DESCRIPTION
**Pull request by jsmartt**
### Description
This PR:
- Adds a `server_profile_template` property to oneview_server_profile
- Adds an `:update_from_template` action to oneview_server_profile
- Updates the ResourceProvider#load_resource method to return the whole resource by default.

Note that in order to get the create_from_template piece to work, I had to split out the creation logic into a separate method, allowing providers to handle creation tasks however they need to without having to override the whole create_or_update and create_if_missing methods.

Also, instead of making ServerProfileTemplateProvider inherit from ServerProfileProvider, I split the common methods into a module and just included them.

I updated the metadata long_description to include the README text, which is a common cookbook pattern, and gives a much more useful description on the Chef Server Manage UI. And finally, bumped the version in the metadata to match the rest of the code/docs, making it ready for a 2.1.0 release. :shipit: 

### Issues Resolved
Fixes #162 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # 2.1.0
+- [#162](https://github.com/HewlettPackard/oneview-chef/issues/162) Add server_profile_template property to oneview_server_profile
+- Add :update_from_template action to oneview_server_profile
```
Comment (tmiotto) [inline CHANGELOG.md]: The CHANGELOG became misconfigured, check it on the 'View' mode.

**Review by tmiotto — CHANGES_REQUESTED** [review]
Fixing it, it should be fine.

**Comment by tmiotto** [comment]
@jsmartt After we merge this PR we are not ready yet to release, there is some changes we want to do in the deployment plan.

See #202

**Comment by jsmartt** [comment]
@tmiotto , I've updated the CHANGELOG. Please let me know if there's anything else to fix

**Review by tmiotto — APPROVED** [review]
Seems fine. :taco:

**Pull request merged**